ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/snmp-settings.j2

```diff
@@ -1,12 +1,27 @@
 {# eos - SNMP Settings#}
 {% if snmp_server is arista.avd.defined %}
 !
+{%     if snmp_server.engine_ids.local is arista.avd.defined %}
+snmp-server engineID local {{ snmp_server.engine_ids.local }}
+{%     endif %}
 {%     if snmp_server.contact is arista.avd.defined %}
 snmp-server contact {{ snmp_server.contact }}
 {%     endif %}
 {%     if snmp_server.location is arista.avd.defined %}
 snmp-server location {{ snmp_server.location }}
 {%     endif %}
+{%     if snmp_server.engine_ids.remotes is arista.avd.defined %}
+{%         for engine_id in snmp_server.engine_ids.remotes %}
+{%             if engine_id.id is arista.avd.defined and engine_id.address is arista.avd.defined %}
+{%                 set remote_engine_ids_cli =  "snmp-server engineID remote " ~ engine_id.address %}
+{%                 if engine_id.udp_port is arista.avd.defined %}
+{%                     set remote_engine_ids_cli =  remote_engine_ids_cli ~ " udp-port " ~  engine_id.udp_port %}
+{%                 endif %}
+{%                 set remote_engine_ids_cli = remote_engine_ids_cli ~ " " ~ engine_id.id %}
+{{ remote_engine_ids_cli }}
+{%             endif %}
+{%         endfor %}
+{%     endif %}
```

Comment:
Please move this block line13-24 below line143(snmp_server.users) to match the cli output.

Reply:
![image](https://user-images.githubusercontent.com/88734745/166937858-b48414c6-c8cf-4c25-8f5f-b10e52520d67.png)

Reply:
Actually it will be more complex than this as it needs to be between local and remote users:
```
Leaf1#show run | se snm
snmp-server engineID local 424242424242424242
snmp-server contact test
snmp-server user test3 test-group v3
snmp-server engineID remote 42.42.42.42 BABABABABABAB
snmp-server user giza hsajsd remote 1.1.1.1 v3
snmp-server user blah blahg remote 1.1.1.1 udp-port 13
```

Reply:
Oh! Ok. That's no problem.

Reply:
to accommodate for this we would need to loop twice through the users, first print the local users, then the remote engineIDs and then the remote ones.